pkg/aws/aws.go

```diff
@@ -98,26 +101,31 @@ func New(config *Config) (*AWS, error) {
 	for _, service := range services {
 		switch service {
 		case "S3":
-			// There are two scenarios:
-			// 1. Running locally, the user must pass in a region and profile to use
-			// 2. Running within an EC2 instance and the region and profile can be derived
-			// I'm going to use the AWS SDK to handle this for me. If the user has provided a region and profile, it will use that.
-			// If not, it will use the EC2 instance metadata service to determine the region and credentials.
-			// This is the same logic that the AWS CLI uses, so it should be fine.
-			options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
-			if config.Region != "" {
-				options = append(options, awsconfig.WithRegion(config.Region))
+			var clients []costexplorer.CostExplorer
+			if len(config.Profiles) == 0 {
+				return nil, fmt.Errorf("no profiles provided")
```

Comment:
(suggestion) We probably want to test this, and in order to make it simpler, I'd suggest adding a package variable `var ErrNoProfiles = errors.New("no profiles provided")` (and BTW, better to use `errors.New` instead of `fmt.Errorf` for simple strings). By adding this error then you can use `errors.Is` in the test.